components/nimbus/src/lib.rs

```diff
@@ -339,8 +339,17 @@ pub struct Experiment {
     pub user_facing_name: String,
     pub user_facing_description: String,
     pub is_enrollment_paused: bool,
-    pub bucket_config: BucketConfig,
+    // TODO: With multi-features,
+    // the `probeSets` field was removed, however for
+    // backward compatability, we will serialize an empty
+    // array if it's missing. this should be removed
+    // once multi-feature is implemented
+    // on the server side, and
+    // we do not expect any more `probe_sets`
+    // in any experiments
+    #[serde(default)]
     pub probe_sets: Vec<String>,
```

Comment:
Serde is pretty smart, doesn't it ignore the field if it's not defined here, regardless of whether or not it's in the actual payload we receive?

Reply:
It does! I kept it here only for when we serialize the experiment down to our persisted storage, where it'll end up with `"probeSets": []` on experiments serialized by an old version of the SDK but no `probeSets` at all on newer experiments if we remove the field.

Honestly, I think that's OK, I can't think of how that could go wrong... when we deserialize from persisted storage, if we deserialize an "old" experiment, we will just ignore the "probeSets" and that should be OK.. It'll also save us the tech debt. I'll give it a shot and add a couple of tests to our `persistence` tests